File: swift/GildedRoseTest.swift

import XCTest

class GildedRoseTest: XCTestCase {
    
    func testFoo() {
        let items = [Item(name: "foo", sellIn: 0, quality: 0)]
        let app = GildedRose(items: items);
        app.updateQuality();
        XCTAssertEqual("fixme", app.items[0].name);
    }
    
}

File: go/gilded-rose_test.go
package main

import "testing"

func Test_Foo(t *testing.T) {
	var items = []*Item{
		&Item{"fixme", 1, 10},
	}

	UpdateQuality(items)

	if items[0].name != "fixme" {
		t.Errorf("Name: Expected %s but got %s ", "fixme", items[0].name)
	}
}
